modules/internal/ExternalArray.chpl

```diff
@@ -349,11 +91,25 @@ module ExternalArray {
     if (arr.domain.low != 0) {
       halt("cannot return an array when the lower bounds is not 0");
     }
-    var externalArr = chpl_make_external_array(c_sizeof(arr.eltType),
-                                               arr.size: uint);
-    chpl__uncheckedArrayTransfer(makeArrayFromExternArray(externalArr,
-                                                          arr.eltType),
-                                 arr);
+    var externalArr = chpl_make_external_array_ptr_free(c_ptrTo(arr[0]),
+                                                        arr.size: uint);
+    arr.externArr = true;
+    arr._borrowed = true;
```

Comment:
This part deserves a comment about how you're intentionally adjusting arr to avoid freeing the buffer since you sent it to the returned chpl_external_array